migration/migrator_validate.go

```diff
@@ -138,3 +139,130 @@ func (m *StateMigrator) ValidateMigratedState(mptRoot common.Hash, zkRoot common
 
 	return nil
 }
+
+func (m *StateMigrator) ValidateNewState(num uint64, mptRoot common.Hash, stateChanges *core.StateChanges) error {
+	if num == 0 {
+		return fmt.Errorf("block number must bigger than 0")
+	}
+
+	parent := m.backend.BlockChain().GetHeaderByNumber(num - 1)
+	if parent == nil {
+		return fmt.Errorf("failed to get parent header by number %d", num-1)
+	}
+	header := m.backend.BlockChain().GetHeaderByNumber(num)
+	if header == nil {
+		return fmt.Errorf("failed to get header by number %d", num)
+	}
+	zkt, err := trie.NewZkMerkleStateTrie(parent.Root, m.zktdb)
+	if err != nil {
+		return fmt.Errorf("failed to create zk state trie: %w", err)
+	}
+	mpt, err := trie.NewStateTrie(trie.StateTrieID(mptRoot), m.mptdb)
+	if err != nil {
+		return fmt.Errorf("fail to create state trie: %w", err)
+	}
+
+	addresses := make(map[common.Address]struct{})
+	slots := make(map[common.Address][]common.Hash)
+
+	// Collect all addresses and slots
+	for addr := range stateChanges.Destruct {
+		addresses[addr] = struct{}{}
+	}
+	for hk := range stateChanges.Accounts {
+		preimage, err := m.readZkPreimage(hk)
+		if err != nil {
+			return fmt.Errorf("failed to read account key preimage: %s", hk)
+		}
+		addr := common.BytesToAddress(preimage)
+		addresses[addr] = struct{}{}
+
+		storageChanges := stateChanges.Storages[hk]
+		for hk := range storageChanges {
+			preimage, err := m.readZkPreimage(hk)
+			if err != nil {
+				return fmt.Errorf("failed to read %s slot key preimage: %s", addr, hk)
+			}
+			slot := common.BytesToHash(preimage)
+			slots[addr] = append(slots[addr], slot)
+		}
+	}
+
+	// Begin the validation process with the collected addresses and storage slots.
+	for addr := range addresses {
+		mptAcc, err := mpt.GetAccount(addr)
+		if err != nil {
+			return fmt.Errorf("failed to get mpt account %s: %w", addr, err)
+		}
+
+		if mptAcc == nil {
+			// If the MPT account does not exist, it is also deleted from ZKT.
+			err = zkt.DeleteAccount(addr)
```

Comment:
It’s not necessary. 
Since the account is deleted, the state is properly updated, and it’s a temporary trie, there’s no need to remove any remaining storage.